atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/mcas/MarkAndCasConsensusForgettingStore.java

```diff
@@ -87,18 +88,22 @@ public void mark(Set<Cell> cells) {
      */
     @Override
     public void atomicUpdate(Cell cell, byte[] value) throws KeyAlreadyExistsException {
-        autobatcher.apply(ImmutableCasRequest.of(cell, inProgressMarkerBuffer, ByteBuffer.wrap(value)));
+        try {
+            autobatcher
+                    .apply(ImmutableCasRequest.of(cell, inProgressMarkerBuffer, ByteBuffer.wrap(value)))
+                    .get();
+        } catch (Exception ex) {
+            if (ex.getCause() instanceof KeyAlreadyExistsException) {
+                throw (KeyAlreadyExistsException) ex.getCause();
+            }
+            throw new SafeRuntimeException("Could not successfully execute atomic update.", ex);
+        }
     }
 
-    /**
-     * This call serially delegated to {@link MarkAndCasConsensusForgettingStore#atomicUpdate(Cell, byte[])}
-     * and does not guarantee atomicity across cells.
-     * The MCAS calls to KVS are batched and hence, in practice, it is possible that the group of cells is served
-     * atomically.
-     * */
     @Override
-    public void atomicUpdate(Map<Cell, byte[]> values) throws KeyAlreadyExistsException {
-        values.forEach(this::atomicUpdate);
+    public void batchAtomicUpdate(Map<Cell, byte[]> values) throws KeyAlreadyExistsException {
+        throw new UnsupportedOperationException("MarkAndCasCFS does not support batch updates currently. Reaching "
```

Comment:
Why don't we support batch updates currently? Could we add that `why` to the exception message?

Reply:
`Batch atomic updates are currently not supported as it's not used or needed, as batching is performed within this class rather than at a higher level. This is due to the fact that we batch both the atomicUpdate and checkAndTouch methods together, which has to be done in this class as this isn't true for all other classes.`

Reply:
IDK if we want to have `PueConsensusForgettingStore` in there, as that could become stale at some point.